src/replica/replica_learn.cpp

```diff
@@ -1507,6 +1508,9 @@ void replica::on_add_learner(const group_check_request &request)
 error_code replica::apply_learned_state_from_private_log(learn_state &state)
 {
     bool duplicating = is_duplicating();
+    // step_back means this round of learn must have been stepped back to include all the
+    // unconfirmed when duplicating. default is false
```

Comment:
In current code, ```duplicating && state.__isset.learn_start_decree && state.learn_start_decree < _app->last_committed_decree() + 1``` means `step_back = true`, I think you should add more comments to describe it. I still can not know the meaning of `step_back` clearly.

Reply:
updateed, and maybe you can see all comment: https://github.com/XiaoMi/rdsn/pull/845/files#diff-304228a830e17d7be16721c87b13698c7d767c47e74c3947e2d005fc30f099b5R1511-R1529